packages/diagrams/frontend/sirius-components-diagrams/src/renderer/edge/MultiLabelEdge.tsx

```diff
@@ -103,19 +114,44 @@ export const MultiLabelEdge = memo(
         ) : null}
         <EdgeLabelRenderer>
           {beginLabel && (
-            <div style={labelContainerStyle(`${sourceLabelTranslation} translate(${sourceX}px,${sourceY}px)`)}>
-              <Label diagramElementId={id} label={beginLabel} faded={!!faded} />
-            </div>
+            <Draggable
+              position={{ x: beginLabel.position.x, y: beginLabel.position.y }}
+              positionOffset={{ x: sourceX, y: sourceY }}
+              onStop={(_e, eventData) => onStop(_e, eventData, 'begin')}
+              scale={zoom}
+              nodeRef={beginLabelNodeRef}>
+              <div
+                style={labelContainerStyle(`${sourceLabelTranslation} translate(${sourceX}px,${sourceY}px)`)}
+                ref={beginLabelNodeRef}>
```

Comment:
Is this normal to have `ref` here?

Reply:
Yes, it's for Draggable API
Look at [nodeRef](https://www.npmjs.com/package/react-draggable#draggable-props)